pennylane_snowflurry/monarq_device.py

```diff
@@ -0,0 +1,142 @@
+    """
+
+    name = "CalculQCDevice"
+    short_name = "monarq.default"
+    pennylane_requires = ">=0.30.0"
+    author = "CalculQuebec"
+    
+    realm = "calculqc"
+    circuit_name = "test circuit"
+    project_id = ""
+    machine_name = "yamaska"
+
+    observables = {
+        "PauliZ"
+    }
+    
+    _processing_config : ProcessingConfig
+    
+    def __init__(self, 
+                 wires = None, 
+                 shots = None,  
+                 client : ApiClient = None,
+                 processing_config : ProcessingConfig = None) -> None:
+
+        if isinstance(shots, int) and (shots < 1 or shots > 1000) or isinstance(shots, list) and (len(shots) < 1 or len(shots) > 1000) or shots == None:
+            raise DeviceException("The number of shots must be contained between 1 and 1000")
+            
+        super().__init__(wires=wires, shots=shots)
+        if client is None:
+            raise DeviceException("The client has not been defined. Cannot establish connection with MonarQ.")
+        
+        self.client = client
+        if not processing_config:
+            processing_config = MonarqDefaultConfig()
+        
+        ApiAdapter.initialize(client)
+        
+        self._processing_config = processing_config
+    
+
+    @property
+    def name(self):
+        return MonarqDevice.short_name
+    
+    
+    def preprocess(
+        self,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Tuple[TransformProgram, ExecutionConfig]:
+        """This function defines the device transfrom program to be applied and an updated execution config.
+
+        Args:
+            execution_config (Union[ExecutionConfig, Sequence[ExecutionConfig]]): A data structure describing the
+            parameters needed to fully describe the execution.
+
+        Returns:
+            TransformProgram: A transform program that when called returns QuantumTapes that the device
+            can natively execute.
+            ExecutionConfig: A configuration with unset specifications filled in.
+        """
+        config = execution_config
+
+        transform_program = TransformProgram()
+        processor = PreProcessor.get_processor(self._processing_config, self.wires)
+        transform_program.add_transform(transform=transform(processor))
+        return transform_program, config
+
+
+    def execute(self, circuits: QuantumTape | list[QuantumTape], execution_config : ExecutionConfig = DefaultExecutionConfig):
+        """
+        This function runs provided quantum circuit on MonarQ
+        A job is first created, and then ran. 
+        Results are then post-processed and returned to the user.
+        """
+        is_single_circuit : bool = isinstance(circuits, QuantumScript)
+        if is_single_circuit:
+            circuits = [circuits]
+        
+         # Check if execution_config is an instance of ExecutionConfig
+        if isinstance(execution_config, ExecutionConfig):
```

Comment:
raise a DeviceException that specifies that ExecutionConfigs are not yet supported for this device instead